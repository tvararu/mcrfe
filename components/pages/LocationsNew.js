import gql from "graphql-tag";
import Query from "../elements/Query";
import GovukMain from "../govuk/Main";
import GovukBreadcrumbs from "../govuk/Breadcrumbs";

const allProvidersQuery = gql`
  query allProviders($providerCode: String!) {
    allProviders(condition: { providerCode: $providerCode }) {
      nodes {
        providerName
      }
    }
  }
`;

const Content = ({ sites }) => null;

export default ({ providerCode }) => (
  <Query query={allProvidersQuery} variables={{ providerCode }}>
    {({ data }) => {
      const provider = data.allProviders.nodes[0];
      return (
        <>
          <GovukBreadcrumbs
            crumbs={[
              { href: "/", text: "Organisations" },
              {
                as: `/organisations/${providerCode}`,
                href: `/organisations?providerCode=${providerCode}`,
                text: provider.providerName
              },
              {
                as: `/organisations/${providerCode}/locations`,
                href: `/organisations/locations?providerCode=${providerCode}`,
                text: "Locations"
              }
            ]}
          />
          <GovukMain>
            <h1 className="govuk-heading-xl">Add a location</h1>
            <Content />
          </GovukMain>
        </>
      );
    }}
  </Query>
);
