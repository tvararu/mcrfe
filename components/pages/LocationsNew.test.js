import renderer from "react-test-renderer";
import LocationsNew from "./LocationsNew";

jest.mock("../elements/Query", () => ({ children }) =>
  children({
    data: {
      allProviders: {
        nodes: [
          {
            providerName: "Acme"
          }
        ]
      }
    }
  })
);

describe("LocationsNew", () => {
  it("renders correctly", () => {
    const component = renderer.create(<LocationsNew providerCode="A01" />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
