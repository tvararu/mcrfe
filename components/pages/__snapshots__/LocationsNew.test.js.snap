// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationsNew renders correctly 1`] = `
Array [
  <nav
    className="govuk-breadcrumbs"
  >
    <ol
      className="govuk-breadcrumbs__list"
    >
      <li
        className="govuk-breadcrumbs__list-item"
      >
        <a
          className="govuk-breadcrumbs__link"
        >
          Organisations
        </a>
      </li>
      <li
        className="govuk-breadcrumbs__list-item"
      >
        <a
          className="govuk-breadcrumbs__link"
        >
          Acme
        </a>
      </li>
      <li
        className="govuk-breadcrumbs__list-item"
      >
        <a
          className="govuk-breadcrumbs__link"
        >
          Locations
        </a>
      </li>
    </ol>
  </nav>,
  <main
    className="govuk-main-wrapper"
    id="main-content"
    role="main"
  >
    <title>
      Add a location - Publish teacher training courses - GOV.UK
    </title>
    <h1
      className="govuk-heading-xl"
    >
      Add a location
    </h1>
    <div
      className="govuk-grid-row"
    >
      <div
        className="govuk-grid-column-two-thirds"
      >
        <form
          action="/organisations/A01/locations"
          method="post"
        >
          <div
            className="govuk-form-group"
          >
            <label
              className="govuk-label"
              htmlFor="site_location_name"
            >
              <h2
                className="govuk-heading-m"
              >
                Name
              </h2>
            </label>
            <input
              className="govuk-input govuk-!-width-full"
              id="site_location_name"
              name="location_name"
              type="text"
            />
          </div>
          <h2
            className="govuk-heading-m govuk-!-margin-top-8"
          >
            Address
          </h2>
          <div
            className="govuk-form-group"
          >
            <label
              className="govuk-label"
              htmlFor="site_address1"
            >
              Building and street
              <span
                className="govuk-visually-hidden"
              >
                line 1 of 2
              </span>
            </label>
            <input
              className="govuk-input govuk-!-width-full"
              id="site_address1"
              name="address1"
              type="text"
            />
          </div>
          <div
            className="govuk-form-group"
          >
            <label
              className="govuk-label"
              htmlFor="site_address2"
            >
              <span
                className="govuk-visually-hidden"
              >
                line 2 of 2
              </span>
            </label>
            <input
              className="govuk-input govuk-!-width-full"
              id="site_address2"
              name="address2"
              type="text"
            />
          </div>
          <div
            className="govuk-form-group"
          >
            <label
              className="govuk-label"
              htmlFor="site_address3"
            >
              Town or city
            </label>
            <input
              className="govuk-input govuk-!-width-full"
              id="site_address3"
              name="address3"
              type="text"
            />
          </div>
          <div
            className="govuk-form-group"
          >
            <label
              className="govuk-label"
              htmlFor="site_address4"
            >
              County
            </label>
            <input
              className="govuk-input govuk-!-width-full"
              id="site_address4"
              name="address4"
              type="text"
            />
          </div>
          <div
            className="govuk-form-group"
          >
            <label
              className="govuk-label"
              htmlFor="site_postcode"
            >
              Postcode
            </label>
            <input
              className="govuk-input form-control-small"
              id="site_postcode"
              name="postcode"
              type="text"
            />
          </div>
          <div
            className="govuk-form-group"
          >
            <label
              className="govuk-label"
              htmlFor="site_region_code"
            >
              Region of UK
            </label>
            <select
              className="govuk-select"
              id="site_region_code"
              name="region_code"
            >
              <option
                value={0}
              >
                London
              </option>
              <option
                value={1}
              >
                South East
              </option>
              <option
                value={2}
              >
                South West
              </option>
              <option
                value={3}
              >
                Wales
              </option>
              <option
                value={4}
              >
                West Midlands
              </option>
              <option
                value={5}
              >
                East Midlands
              </option>
              <option
                value={6}
              >
                Eastern
              </option>
              <option
                value={7}
              >
                North West
              </option>
              <option
                value={8}
              >
                Yorkshire and the Humber
              </option>
              <option
                value={9}
              >
                North East
              </option>
              <option
                value={10}
              >
                Scotland
              </option>
              <option
                value={11}
              >
                No Region
              </option>
            </select>
          </div>
          <input
            className="govuk-button"
            type="submit"
            value="Save"
          />
        </form>
        <p
          className="govuk-body"
        >
          <a
            className="govuk-link"
          >
            Cancel changes
          </a>
        </p>
      </div>
    </div>
  </main>,
]
`;
